Mathlib/Data/List/Sort.lean

```diff
@@ -157,6 +157,27 @@ theorem Sorted.rel_of_mem_take_of_mem_drop {l : List α} (h : List.Sorted r l) {
   exact h.rel_nthLe_of_lt _ _ (Nat.lt_add_right _ (lt_min_iff.mp hix).left)
 #align list.sorted.rel_of_mem_take_of_mem_drop List.Sorted.rel_of_mem_take_of_mem_drop
 
+theorem Sorted.filter {l : List α} (f : α → Bool) (h : Sorted r l) :
+    Sorted r (filter f l) := by
```

Comment:
This follows from more general facts:

- (not yet in `Mathlib`) a sublist of a sorted list is sorted;
- `filter f l` is a sublist of `l`, see [List.filter_sublist](https://leanprover-community.github.io/mathlib4_docs/Std/Data/List/Lemmas.html#List.filter_sublist).

Reply:
I've moved this to #18466